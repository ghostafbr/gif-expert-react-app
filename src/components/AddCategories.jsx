import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (evt) => {
    evt.preventDefault();
    const inputValueTrimed = inputValue.trim();
    if (inputValueTrimed.length <= 1) return;

    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValueTrimed);
    setInputValue('');
  };

  return (
    <>
      <form onSubmit={onSubmit} aria-label='form'>
        <input
          type='text'
          placeholder='Search gif'
          value={inputValue}
          onChange={onInputChange}
        />
      </form>
    </>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
