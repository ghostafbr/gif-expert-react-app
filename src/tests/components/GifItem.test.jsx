import { render, screen } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Pruebas en <GifItem />', () => {
  const title = 'Vegeta';
  const url = 'https://vegetal.com/vegeta.jpg';

  test('should match with the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('must show the image with the correct url and alt', () => {
    render(<GifItem title={title} url={url} />);
    /* expect(screen.getByRole('img').src).toBe(url);*/

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test('must show the title in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
