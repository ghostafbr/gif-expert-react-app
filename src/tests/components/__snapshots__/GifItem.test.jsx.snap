// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Vegeta"
      src="https://vegetal.com/vegeta.jpg"
    />
    <p>
      Vegeta
    </p>
  </div>
</div>
`;
